### DISCUSSION TITLE
Find path with weight condition

### DISCUSSION THREAD

**Discussion by Volodimirich** [networkx/networkx#4697] [Q&A]
For example, i have a weighted graph G, and I want to find a path between two nodes, with edge capacity greater, than some value.
Library path algorithms find a path without considering without path condition. 

Sorry for stupid question, but i didn't find the answer for it :/

**Comment by dschult** [comment]
NetworkX has shortest path functions. But it also has flow functions which deal with capacities and node demands etc. Maybe with some more information we can steer you to the right approach.

What are you trying to do?  You say you are trying to find a path with specified capacity or greater. A depth first search might do this. Tell us more.

**Reply by Volodimirich** [reply]
Thanks for answer!
I am trying to simulate adding requests to the network schedule.
For example, i have some graph G:

```
G = nx.Graph()
G.add_nodes_from([0, 1, 2, 3])
G.add_edges_from([(0,1), (1, 2), (2, 3), (0, 3)])
nx.set_edge_attributes(G, values=3, name='capacity')

```

And for example, i try to find path in this graph from 0 to 3 with capacity 2
The expected path is [0,3]. After that i change edge capacity between 0 and 3:

`G[0][3]['capacity'] -= 2`

And after that i am trying to find a new path between 0 and 3 with capacity 2, and I want to get next path: [0, 1, 2, 3] (because G[0][3]['capacity'] = 1).

How can i get it?

**Comment by MridulS** [comment]
@Volodimirich I may have understood this wrong but if you want to find the shortest path only on edges with a certain capacity you could use `subgraph_view` and a `edge_filter` to filter out the edges and then calculate the shortest path.

``` python
In [1]: import networkx as nx

In [2]: G = nx.Graph()
   ...: G.add_nodes_from([0, 1, 2, 3])
   ...: G.add_edges_from([(0,1), (1, 2), (2, 3), (0, 3)])
   ...: nx.set_edge_attributes(G, values=3, name='capacity')

In [3]: nx.shortest_path(G, 0, 3)
Out[3]: [0, 3]

In [4]: G[0][3]['capacity'] -= 2

In [5]: cap = 2 # change this to control the min capacity of the network
   ...: def capacity_filter(u, v):
   ...:     return G[u][v]['capacity'] >= cap
   ...:

In [6]: valid_graph = nx.subgraph_view(G,  filter_edge=capacity_filter)

In [7]: nx.shortest_path(valid_graph, 0, 3)
Out[7]: [0, 1, 2, 3]
```
